### PR TITLE
No warn-unused:params for unimplemented method

### PR DESCRIPTION
**Pull request by som-snytt**
Cut some slack for `def f(i: Int) = ???`.

### CODE REVIEW

**Comment by SethTisue** [comment]
needs rebase

**Comment by som-snytt** [comment]
Showing how even a half-LOC is a cost.

Conflict due to moors pushes, showing that it's not really possible to manage conflicts over multipled PRs in advance of merges.

File: test/files/neg/warn-unused-params.scala [hunk 1]
```diff
@@ -67,3 +67,7 @@ class Reusing(u: Int) extends Unusing(u)   // no warn
 class Main {
   def main(args: Array[String]): Unit = println("hello, args")  // no warn
 }
+
+trait Unimplementation {
+  def f(u: Int): Int = ???        // no warn for param in unimplementation
+}
```
Comment (hrhino) [inline test/files/neg/warn-unused-params.scala:73]: Pedantically, wouldn't an "unimplementation" be when you override a concrete method like this?

**Review by hrhino — APPROVED** [review]
There was a ticket once about how hard it is to use a custom `Predef`. Probably Paul's. I suppose this just adds on to the complaint.

👍 though, anything that helps make the unused warnings more reliably accurate is a boon.

File: test/files/neg/warn-unused-params.scala (hunk 1 above)
Comment (som-snytt) [reply test/files/neg/warn-unused-params.scala:73]: "Deimplementation." But I see it's `NotImplementedError` and not `UnimplementedException`. Unfortunately, it is final.

**Comment by adriaanm** [comment]
One rebase pays admission to 2.12.6. What could possibly go wrong???

**Review by adriaanm — APPROVED** [review]

**Comment by som-snytt** [comment]
@adriaanm ha, I just saw what you did???

I already warned someone jonesing for 2.12.7 that you might need to pull that version out of reserve at any moment. Version, like age, is just a dotted number.

File: test/files/neg/warn-unused-params.scala (hunk 1 above)
Comment (som-snytt) [reply test/files/neg/warn-unused-params.scala:73]: Also, the more modest April 1 post would have announced that Seth had persuaded Martin to rename `NotImplementedError` to `UnimplementedMistake`, and how this represented a huge advance in the face of Martin's conservative approach to language design and backward compatibility. It took a long session at the Biergarten and all of Seth's charms. Basically he drank him under the table and agreed not to post anything on twitter.

Comment (hrhino) [reply test/files/neg/warn-unused-params.scala:73]: > It took a long session at the Biergarten and all of Seth's charms. Basically he drank him under the table and agreed not to post anything on twitter.

And that's _not_ how we got Setha?

**Pull request merged**